src/app/child-dev-project/attendance/add-day-attendance/roll-call-setup/roll-call-setup.component.spec.ts

```diff
@@ -3,51 +3,51 @@ import {
   fakeAsync,
   flush,
   TestBed,
-  waitForAsync,
 } from "@angular/core/testing";
 
 import { RollCallSetupComponent } from "./roll-call-setup.component";
-import { FormDialogModule } from "../../../../core/form-dialog/form-dialog.module";
 import { EntityMapperService } from "../../../../core/entity/entity-mapper.service";
 import { SessionService } from "../../../../core/session/session-service/session.service";
 import { User } from "../../../../core/user/user";
-import { AttendanceService } from "../../attendance.service";
 import { RecurringActivity } from "../../model/recurring-activity";
 import { Note } from "../../../notes/model/note";
+import { ChildrenService } from "../../../children/children.service";
+import { AttendanceModule } from "../../attendance.module";
+import { MockDatabase } from "../../../../core/database/mock-database";
+import { Database } from "../../../../core/database/database";
+import { MatNativeDateModule } from "@angular/material/core";
 
 describe("RollCallSetupComponent", () => {
   let component: RollCallSetupComponent;
   let fixture: ComponentFixture<RollCallSetupComponent>;
 
   let mockEntityService: jasmine.SpyObj<EntityMapperService>;
-  let mockAttendanceService: jasmine.SpyObj<AttendanceService>;
+  let mockChildrenService: jasmine.SpyObj<ChildrenService>;
 
-  beforeEach(
-    waitForAsync(() => {
-      mockEntityService = jasmine.createSpyObj("mockEntityService", [
-        "save",
-        "loadType",
-      ]);
-      mockEntityService.loadType.and.resolveTo([]);
-      mockAttendanceService = jasmine.createSpyObj("mockAttendanceService", [
-        "getEventsOnDate",
-      ]);
-      mockAttendanceService.getEventsOnDate.and.resolveTo([]);
+  beforeEach(() => {
+    mockEntityService = jasmine.createSpyObj("mockEntityService", [
+      "save",
+      "loadType",
+    ]);
+    mockEntityService.loadType.and.resolveTo([]);
+
+    mockChildrenService = jasmine.createSpyObj(["queryRelationsOf"]);
+    mockChildrenService.queryRelationsOf.and.resolveTo([]);
 
-      TestBed.configureTestingModule({
-        declarations: [RollCallSetupComponent],
-        imports: [FormDialogModule],
-        providers: [
-          { provide: EntityMapperService, useValue: mockEntityService },
-          {
-            provide: SessionService,
-            useValue: { getCurrentUser: () => new User("") },
-          },
-          { provide: AttendanceService, useValue: mockAttendanceService },
-        ],
-      }).compileComponents();
-    })
-  );
+    TestBed.configureTestingModule({
+      declarations: [RollCallSetupComponent],
+      imports: [AttendanceModule, MatNativeDateModule],
+      providers: [
+        { provide: EntityMapperService, useValue: mockEntityService },
+        {
+          provide: SessionService,
+          useValue: { getCurrentUser: () => new User("") },
+        },
+        { provide: Database, useClass: MockDatabase },
```

Comment:
any particular reason to go away from mocking the direct dependency `AttendanceService` and instead providing the database itself?

Reply:
The component uses the factory method for creating notes from activities. This function now moved to the AttendanceService and it didn't really make sense to mock this method because